sklearn/datasets/tests/test_openml.py

```diff
@@ -1523,3 +1524,25 @@ def test_missing_values_pandas(monkeypatch):
     # there are nans in the categorical
     assert penguins.data["sex"].isna().any()
     assert_array_equal(cat_dtype.categories, ["FEMALE", "MALE", "_"])
+
+
+def test_open_openml_url_retry_on_network_error(monkeypatch):
+    def _mock_urlopen_network_error(request, *args, **kwargs):
+        raise HTTPError("", 404, "Simulated network error", None, None)
+
+    monkeypatch.setattr(
+        sklearn.datasets._openml, "urlopen", _mock_urlopen_network_error
+    )
+
+    invalid_openml_url = "invalid-url"
+
+    with pytest.warns(
+        UserWarning,
+        match=(
```

Comment:
Apparently it's not necessary here, but sometimes it might be necessary to use:

```suggestion
        match=re.escape(
```

to avoid having specific characters be interpreted as regexp operators.

Reply:
Did not know about that, i'll change it right now. Thank you !